apps/meteor/ee/app/livechat-enterprise/server/lib/Helper.js

```diff
@@ -199,65 +205,17 @@ export const updateQueueInactivityTimeout = () => {
 	});
 };
 
-export const updateRoomPriorityHistory = (rid, user, priority) => {
-	const history = {
-		priorityData: {
-			definedBy: user,
-			priority: priority || {},
-		},
-	};
-
-	Messages.createPriorityHistoryWithRoomIdMessageAndUser(rid, '', user, history);
-};
-
-export const updateInquiryQueuePriority = (roomId, priority) => {
-	const inquiry = LivechatInquiry.findOneByRoomId(roomId, { fields: { rid: 1, ts: 1 } });
-	if (!inquiry) {
+export const updateSLAInquiries = async (sla) => {
+	if (!sla) {
 		return;
 	}
 
-	let { ts: estimatedServiceTimeAt } = inquiry;
-	let queueOrder = 1;
-	let estimatedWaitingTimeQueue = 0;
-
-	if (priority) {
-		const { dueTimeInMinutes } = priority;
-		queueOrder = 0;
-		estimatedWaitingTimeQueue = dueTimeInMinutes;
-		estimatedServiceTimeAt = new Date(estimatedServiceTimeAt.setMinutes(estimatedServiceTimeAt.getMinutes() + dueTimeInMinutes));
-	}
-
-	LivechatInquiry.setEstimatedServiceTimeAt(inquiry.rid, {
-		queueOrder,
-		estimatedWaitingTimeQueue,
-		estimatedServiceTimeAt,
-	});
-};
-
-export const removePriorityFromRooms = async (priorityId) => {
-	const result = await Promise.allSettled(
-		LivechatRoomsRaw.findOpenRoomsByPriorityId(priorityId).forEach((room) => {
-			updateInquiryQueuePriority(room._id);
-		}),
-	);
-	const rejected = result.filter((r) => r.status === 'rejected').map((r) => r.reason);
-	if (rejected.length) {
-		logger.error({ msg: `Error while removing priority from ${rejected.length} rooms`, reason: rejected[0] });
-		logger.debug({ msg: 'Rejection results', rejected });
-	}
-
-	await LivechatRoomsRaw.unsetPriorityByIdFromAllOpenRooms(priorityId);
-};
-
-export const updatePriorityInquiries = (priority) => {
-	if (!priority) {
-		return;
-	}
-
-	const { _id: priorityId } = priority;
-	LivechatRooms.findOpenByPriorityId(priorityId).forEach((room) => {
-		updateInquiryQueuePriority(room._id, priority);
+	const { _id: slaId } = sla;
+	const promises = [];
+	await LivechatRoomsRaw.findOpenBySlaId(slaId).forEach((room) => {
+		promises.push(updateInquiryQueueSla(room._id, sla));
 	});
+	await Promise.allSettled(promises.length ? promises : []);
```

Comment:
this is the same right?
```suggestion
	await Promise.allSettled(promises);
```

Reply:
I was probably on drugs these day